src/pages/createConfig/step2AddConfig/step2AddConfig.tsx

```diff
@@ -0,0 +1,77 @@
+import { Box, Typography } from '@mui/material';
+import { languages } from 'monaco-editor';
+import ajv from 'ajv';
+import { useQuery } from '@tanstack/react-query';
+import { getSchema } from '../../../api/client';
+import { MonacoEditor } from '../../../components/monacoEditor/monacoEditor';
+import { useCallback, useEffect, useMemo } from 'react';
+import { useMonaco } from '@monaco-editor/react';
+import { ConfigData } from '../createConfig.types';
+
+type Step2AddConfigProps = {
+  schemaId: string;
+  onDataChange: (data: ConfigData | undefined, isValid: boolean) => void;
+  onJsonStringChange: (json: string | undefined) => void;
+  initialJsonStringData?: string | undefined;
+};
+
+export const Step2AddConfig: React.FC<Step2AddConfigProps> = ({ onDataChange, onJsonStringChange, schemaId, initialJsonStringData }) => {
+  const fetchSchema = useCallback(() => getSchema({ id: schemaId, shouldDereference: true }), [schemaId]);
+
+  const { data: schema, isSuccess } = useQuery({
+    queryKey: ['getSchema2'],
+    queryFn: fetchSchema,
+  });
+
+  const monaco = useMonaco();
+
+  const ajvInstance = useMemo(() => new ajv(), []);
```

Comment:
you can initialise it outside of the component

Reply:
done